Doc/library/doctest.rst

```diff
@@ -381,12 +398,20 @@ The fine print:
 What's the Execution Context?
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-By default, each time :mod:`doctest` finds a docstring to test, it uses a
-*shallow copy* of :mod:`M`'s globals, so that running tests doesn't change the
-module's real globals, and so that one test in :mod:`M` can't leave behind
-crumbs that accidentally allow another test to work.  This means examples can
-freely use any names defined at top-level in :mod:`M`, and names defined earlier
-in the docstring being run. Examples cannot see names defined in other
+Within a docstring, later examples can use names defined by earlier
+examples. It's fine for an example to set up state, and
+have no output. 
```

Comment:
IMO this paragraph is unneeded.  It is already clearly demonstrated in your enhanced example.